### PR TITLE
improve `ids_to_indices` docstring

### PR DESCRIPTION
**Pull request by h-mayorquin**
From the discussion of #2280.

I am not sure I would like to add a check @alejoe91 . If someone passes and int or a string we could coerce that to a list with one element but I dislike that as well.

### CODE REVIEW

**Review by alejoe91 — DISMISSED** [review]

**Comment by alejoe91** [comment]
@h-mayorquin I would add an 
```
assert isinstance(ids, list), "..."
```
What do you think?

**Comment by h-mayorquin** [comment]
To be honest I was kind of lazy to think about the type. Can it be an array as well or only a list? I suspect some of the code might be passing np.arrays and that assert will break something else. I can deal with that but I would rather do that in another PR immediatley after that just to get the docstring improvement immediatley.

File: src/spikeinterface/core/base.py [hunk 1]
```diff
@@ -90,17 +90,27 @@ def _check_segment_index(self, segment_index: Optional[int] = None) -> int:
...
-        into a indices array.
-        Useful to manipulate:
-          * data
-          * properties
+        Convert a list of IDs into indices, either as an array or a slice.
 
-        "prefer_slice" is an efficient option that tries to make a slice object
-        when indices are consecutive.
+        This function is designed to transform a list of IDs (such as channel or unit IDs) into an array of indices.
+        These indices are useful for data manipulation and accessing properties. When `prefer_slice` is set to `True`,
```
Comment (zm711) [inline src/spikeinterface/core/base.py:98]: I think data manipulation is a little loaded a term for scientific research. Maybe data interaction instead? 

I've been slowly trying to change references to `fake data` in the documentation to `simulated data` for the same reason.

Comment (zm711) [reply src/spikeinterface/core/base.py:98]: Or interacting with the data (if you prefer that flow better).

Comment (h-mayorquin) [reply src/spikeinterface/core/base.py:98]: x D
That's a good point : )

Comment (h-mayorquin) [inline src/spikeinterface/core/base.py:98]:
```suggestion
        These indices are useful for interacting with data and accessing properties. When `prefer_slice` is set to `True`,
```

**Comment by alejoe91** [comment]
Yeah it could be np.ndarray as well

**Comment by h-mayorquin** [comment]
@alejoe91 let's merge this and I will add add the check in another PR immediatley  just to get the improved docstring in main.

**Comment by JoeZiminski** [comment]
@h-mayorquin this is nice, an important discussion I noticed in some places checks are made against `list` / `np.array` e.g. [here](https://github.com/SpikeInterface/spikeinterface/blob/8e32955e0616b9e399c253aaf0602f3aa646dae4/src/spikeinterface/preprocessing/silence_periods.py#L59) but in that example it could also accept a `tuple`? could this function also accept a `tuple`?

I wonder if it is worth introducing a single function to centralise all of this logic that is currently spread through the codebase. Could be core / utills like:

```
def ensure_array_like(input_):
    
    if instance(str, float, int):  # bit jenky
         array_like = [input_]

    else:
         assert isinstance(input_, (list, tuple, np.ndarray)), ""

    return input_
```
(with specifics up for discussion e.g. is it a good idea to try and coerce to array-like)

**Comment by samuelgarcia** [comment]
Of course np.ndarray can be the input, I think this is the most general case.
We could `ids = np.asarray(ids)` no ?

And so the implementation `indices = np.array([_main_ids.index(id) for id in ids], dtype=int)` could be changed by a np.seachsorted implementation  instead (faster).

**Comment by h-mayorquin** [comment]
@JoeZiminski 
Yes, I am faimliar with the arguments:
> don't repeat yourself (DRY) principle is the number one refactoring that a) reduces bugs b) allows easy extensibiltiy c) actually makes the code more readable (if good function names are used)

Those points makes sense. I think that the problem is that it easily lends itself to programmers eagerly trying to generalize and optimize code (stuff that we like doing) which I think is bad. Here when discussing fixing this we are already discussing adding more features, you see? I think that applying don't repeat yourself to refactor existing code is great (which is what you are proposing) but a bunch of times don't repeat yourself is done in advance of the actual uses cases for the sake of "be prepared" or "future proof" and that tends to be bad (you create an abstraction without use cases which tends to fail because abstracting is hard).

I am not familiar with the specific book recommendation, thanks. Any section of the book that you would recommend specifically?

Best regards,

**Pull request merged**